pipit/util/test_generator.py

```diff
@@ -0,0 +1,366 @@
+import random
+import textwrap
+import pandas as pd
+
+
+class FakeNode:
+    """
+    A single node of the calling tree used to produce fake traces.
+    Represents a single function call.
+    """
+
+    def __init__(
+        self, name, exc_time, mpi_type="", mpi_tgt=0, mpi_volume=0, mpi_time=0
+    ):
+        self.name = name
+        self.exc_time = exc_time
+        self.inc_time = exc_time
+        self.children = {}  # run_time -> child node
```

Comment:
Is run_time simply the time since the start of the parent at which the child executes or is it the exclusive time since the last child/start of the parent?